[Profiler][Trivial] Format profiler_python.cpp

robieta opened the pull request:
Summary: There are some unfortunate style issues, like four space indents and various other minor issues. There is a pretty big overhaul coming to the python tracer, so I want to be able to commit them with more style compliant code.

Test Plan: N/A. Lints only.

Reviewed By: aaronenyeshi

Differential Revision: D36070201

aaronenyeshi reviewed (approved):
Reviewed internally, LGTM.

robieta commented:
@pytorchbot rebase this please

Commit:
[Profiler] Add EventFieldsVisitor (#76931)

Summary:
One source of complexity in profiler_kineto is that we do most things twice: once to set a field in `kineto_events_.back()`, and once for the metadata json. These have historically been chained, with the KinetoEvent used to populate the metadata fields. However this is hard to read and error prone, as we have one giant block of assignments followed by another giant block. It also means that logic about whether a field is present or not is duplicated.

This PR replaces this logic with a visitor that writes both together. E.g.
```
    auto& dtypes = result_.get().inputs_.dtypes_;
    if (!dtypes.empty()) {
      kineto_event_.get().dtypes(dtypes);
      out.emplace_back("Input type", dtypesToStr(dtypes));
    }
```

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76931

Test Plan: Unit tests.

Differential Revision: D36070202

Pulled By: robieta

fbshipit-source-id: 0f5d1ffaeed79044810a2cd8bf7e501acbbdb886

Commit:
[Profiler] Remove `addMemoryUsageActivity` from kineto_shim (#76678)

Summary:
`addMemoryUsageActivity` is effectively the same as `addCPUActivity` except it also has `addMetadata` calls. This change generalizes `addCPUActivity` (We have to define an enum to map to `libkineto::ActivityType`) to accept arbitrary metadata strings.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76678

Test Plan: Unit tests

Reviewed By: aaronenyeshi

Differential Revision: D36070203

Pulled By: robieta

fbshipit-source-id: 89851259a6a5d2fcf530438cfb799fccdd000663

Commit:
[Profiler][Trivial] Format profiler_python.cpp (#76930)

Summary:
There are some unfortunate style issues, like four space indents and various other minor issues. There is a pretty big overhaul coming to the python tracer, so I want to be able to commit them with more style compliant code.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76930

Test Plan: N/A. Lints only.

Differential Revision: D36070201

Pulled By: robieta

fbshipit-source-id: 6c1e66953888464e0d640eb1c567a7c3fe94f56a

robieta commented:
@pytorchbot merge this please
